modules/hdf4_handler/hdfdesc.cc

```diff
@@ -5867,11 +5909,200 @@ bool obtain_eos2_gd_ll_info(const string & fname, const string & grid_name, int3
     if (GCTP_CEA == projcode || GCTP_GEO == projcode)
         oned_ll = true;
 
+    eg_info.projcode = projcode;
+    eg_info.zone = zone;
+    eg_info.sphere = sphere;
+    for (int i = 0; i<2;i++) {
+        eg_info.upleft[i]   = upleft[i];
+        eg_info.lowright[i] = lowright[i];
+    }
+    for (int i = 0; i<16;i++)
+        eg_info.params[i] = params[i];
+  
     GDdetach(gridid);
     GDclose(gfid);
 
     return true;
 }
+void add_dummy_grid_cv(D4Group *d4_grp, const eos2_grid_info_t &eg_info) {
+
+    string dummy_proj_cf_name = "eos5_cf_projection";
+    auto dummy_proj_cf_unique = make_unique<HDFEOS2GeoCFProj>(dummy_proj_cf_name, dummy_proj_cf_name);
+    HDFEOS2GeoCFProj *dummy_proj_cf = dummy_proj_cf_unique.release();
+    dummy_proj_cf->set_is_dap4(true);
+
+    if (eg_info.projcode == GCTP_SNSOID) {
+
+        add_var_dap4_attr(dummy_proj_cf, "grid_mapping_name", attr_str_c, "sinusoidal");
+        add_var_dap4_attr(dummy_proj_cf, "longitude_of_central_meridian", attr_float64_c, "0.0");
+        add_var_dap4_attr(dummy_proj_cf, "earth_radius", attr_float64_c, "6371007.181");
+        add_var_dap4_attr(dummy_proj_cf, "_CoordinateAxisTypes", attr_str_c, "GeoX GeoY");
+    } else if (eg_info.projcode == GCTP_PS)
+        add_ps_cf_grid_mapping_attrs(dummy_proj_cf, eg_info);
+    else if (eg_info.projcode == GCTP_LAMAZ)
+        add_lamaz_cf_grid_mapping_attrs(dummy_proj_cf, eg_info);
+    d4_grp->add_var_nocopy(dummy_proj_cf);
```

Comment:
... you can use release() like this: 
        d4_grp->add_var_nocopy(dummy_proj_cf_unique.release());

The benefit is that if anything throws an exception, the unique_ptr cleans up the memory without the calling code doing anything.

Reply:
Oh, I see. Got it.